Core/source/mobile-interface/format-output.ts

```diff
@@ -116,22 +117,22 @@ export const fmtContentBlock = (allContentBlocks: MsgBlock[]): { contentBlock: M
   return { contentBlock: MsgBlock.fromContent('plainHtml', msgContentAsHtml), text: msgContentAsText.trim() };
 }
 
-export const fmtRes = (response: {}, data?: Buf | Uint8Array): Buffers => {
-  const buffers: Buffers = [];
-  buffers.push(Buf.fromUtfStr(JSON.stringify(response)));
-  buffers.push(Buf.fromUtfStr('\n'));
-  if (data) {
-    buffers.push(data);
-  }
-  return buffers;
+export const fmtRes = (response: {}, data?: Buf | Uint8Array): EndpointRes => {
+  return {
+    json: JSON.stringify(response),
+    data: data || new Uint8Array(0)
+  };
 }
 
-export const fmtErr = (e: any): Buf => Buf.concat(fmtRes({
-  error: {
-    message: String(e),
-    stack: e && typeof e === 'object' ? e.stack || '' : ''
-  }
-}));
+export const fmtErr = (e: any): Buf => {
+  const res = fmtRes({
+    error: {
+      message: String(e),
+      stack: e && typeof e === 'object' ? e.stack || '' : ''
+    }
+  });
+  return Buf.fromUtfStr(res.json);
+}
```

Comment:
Please have this also return `EndpointRes`, so you would just straight `return fmtRes({...`